app/controllers/roster_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RosterController < ApplicationController
```

Comment:
I don't think "roster" is the right name to use here. This controller's job will be to trigger a username update, which is independent of which courses that user is participating in. (We also don't use the term "roster" anywhere else in the system, so it's likely to be confusing.)